src/quota_aggregator_impl_test.cc

```diff
@@ -0,0 +1,354 @@
+  EXPECT_OK(aggregator_->CacheResponse(request1_, pass_response1_));
+  EXPECT_OK(aggregator_->Quota(request1_, &response));
+  EXPECT_TRUE(MessageDifferencer::Equals(response, pass_response1_));
+}
+
+TEST_F(QuotaAggregatorImplTest, TestFirstReqeustQuotaExceed) {
+  AllocateQuotaResponse response;
+
+  EXPECT_ERROR_CODE(Code::NOT_FOUND, aggregator_->Quota(request1_, &response));
+  EXPECT_OK(aggregator_->CacheResponse(request1_, error_response1_));
+  EXPECT_OK(aggregator_->Quota(request1_, &response));
+  EXPECT_TRUE(MessageDifferencer::Equals(response, error_response1_));
+}
+
+TEST_F(QuotaAggregatorImplTest, TestNotMatchingServiceName) {
+  *(request1_.mutable_service_name()) = "some-other-service-name";
+  AllocateQuotaResponse response;
+
+  EXPECT_ERROR_CODE(Code::INVALID_ARGUMENT,
+                    aggregator_->Quota(request1_, &response));
+}
+
+TEST_F(QuotaAggregatorImplTest, TestNoOperation) {
+  request1_.clear_allocate_operation();
+  AllocateQuotaResponse response;
+
+  EXPECT_ERROR_CODE(Code::INVALID_ARGUMENT,
+                    aggregator_->Quota(request1_, &response));
+}
+
+TEST_F(QuotaAggregatorImplTest, TestFlushAggregatedRecord) {
+  AllocateQuotaResponse response;
+
+  EXPECT_ERROR_CODE(Code::NOT_FOUND, aggregator_->Quota(request1_, &response));
+  EXPECT_OK(aggregator_->CacheResponse(request1_, pass_response1_));
+  EXPECT_OK(aggregator_->Quota(request1_, &response));
+  EXPECT_TRUE(MessageDifferencer::Equals(response, pass_response1_));
+
+  EXPECT_EQ(flushed_.size(), 0);
+
+  // simulate refresh timeout
+  std::this_thread::sleep_for(std::chrono::milliseconds(110));
+
+  EXPECT_OK(aggregator_->Flush());
+  EXPECT_EQ(flushed_.size(), 1);
+}
+
+TEST_F(QuotaAggregatorImplTest, TestAggregatedRecordNoRefresh) {
+  AllocateQuotaResponse response;
+
+  EXPECT_ERROR_CODE(Code::NOT_FOUND, aggregator_->Quota(request1_, &response));
+  EXPECT_OK(aggregator_->CacheResponse(request1_, pass_response1_));
+  EXPECT_OK(aggregator_->Quota(request1_, &response));
+  EXPECT_TRUE(MessageDifferencer::Equals(response, pass_response1_));
+
+  EXPECT_OK(aggregator_->Flush());
+  EXPECT_EQ(flushed_.size(), 0);
+}
+
+TEST_F(QuotaAggregatorImplTest, TestCacheRefreshAllAggregated) {
+  AllocateQuotaResponse response1;
+  AllocateQuotaResponse response2;
+
+  EXPECT_OK(aggregator_->CacheResponse(request1_, pass_response1_));
+  std::this_thread::sleep_for(std::chrono::milliseconds(50));
+  EXPECT_OK(aggregator_->CacheResponse(request2_, pass_response2_));
+  EXPECT_OK(aggregator_->Quota(request1_, &response1));
+  EXPECT_OK(aggregator_->Quota(request2_, &response2));
+  EXPECT_TRUE(MessageDifferencer::Equals(response1, pass_response1_));
+  EXPECT_TRUE(MessageDifferencer::Equals(response2, pass_response2_));
+
+  EXPECT_EQ(flushed_.size(), 0);
+
+  // expire request1
+  std::this_thread::sleep_for(std::chrono::milliseconds(60));
+
+  EXPECT_OK(aggregator_->Flush());
+  EXPECT_EQ(flushed_.size(), 1);
```

Comment:
could you verify which item is flushed out
also verify if call Quota() again for that item, what will return.